pkg/fdbstatus/status_checks.go

```diff
@@ -469,3 +478,32 @@ func HasDesiredFaultToleranceFromStatus(log logr.Logger, status *fdbv1beta2.Foun
 
 	return true
 }
+
+// ConfigurationChangeAllowed will return an error if the configuration change is assumed to be unsafe. If no error
+// is returned the configuration change can be applied.
+func ConfigurationChangeAllowed(status *fdbv1beta2.FoundationDBStatus, useRecoveryState bool) error {
```

Comment:
@sbodagala and @jzhou77 are there any other checks that the operator should be performing before trying to change the cluster configuration?

Reply:
I am still getting familiar with this area so I can't comment on what other checks the operator should perform. Checking that the status hasn't returned any error is reasonable. But the changeConfig API itself appears to be doing a number of checks and returns an error (https://github.com/apple/foundationdb/blob/967a546e15e81714e1dd763108b8afc3775416c5/fdbclient/include/fdbclient/GenericManagementAPI.actor.h#L47) if it can't change the configuration. Why are you checking the storage server lag below (I am trying to understand how it is related to/could affect changing the cluster configuration)?